drivers/spi/spi_sam.c

```diff
@@ -180,7 +180,7 @@ static void spi_sam_finish(Spi *regs)
 /* Fast path that transmits a buf */
 static void spi_sam_fast_tx(Spi *regs, const struct spi_buf *tx_buf)
 {
-	struct k_spinlock lock;
+	struct k_spinlock lock = {};
```

Comment:
How about moving the spinlock in `spi_sam_data` (and pass a dev pointer around)? Having the lock on the stack is such a weird pattern, I think we'd better off avoiding it. (cc @andyross)

Reply:
I could, let me know which way you want to go on that, I would need to pipe the data into the function.

Reply:
Yeah I think it'd work out better, especially since it's used in multiple points around the code to protect access to the very same registers.

Reply:
A very reasonable alternative if you don't want to complicate things is just to make the lock a static file-scope variable and share it globally across all users/devices.  That has the same effect that a 1-cpu interrupt lock would produce anyway.  And even in SMP, lock granularity **mostly** doesn't matter, it's only a tiny handful of high-usage devices that really need to split locks (and a smaller handful of impossible-to-split locks like the global scheduler lock that end up being 99.99% of the contention points anyway).

Reply:
Sounds good, I'll update tonight